prow/entrypoint/run.go

```diff
@@ -134,11 +150,35 @@ func (o Options) ExecuteProcess() (int, error) {
 		}
 	}
 
-	executable := o.Args[0]
+	combinedCommandAndArgs := o.CombinedCommandAndArgs()
+	// Set the first arg as entrypoint for now
+	executable := combinedCommandAndArgs[0]
 	var arguments []string
-	if len(o.Args) > 1 {
-		arguments = o.Args[1:]
+	if len(combinedCommandAndArgs) > 1 {
+		arguments = combinedCommandAndArgs[1:]
 	}
+	// Discover default entrypoint from the image only when it's not explicitly
+	// defined in the job
+	if o.UseDefaultEntrypoint != nil && *o.UseDefaultEntrypoint && len(o.Command) == 0 && o.ImageRef != "" {
+		img, err := remote.Image(o.ImageRef,
+			remote.WithContext(context.Background()),
+			remote.WithAuthFromKeychain(keychain))
+		if err != nil {
+			return err
+		}
+
+		cfg, err := img.ConfigFile()
+		if err != nil {
+			return err
+		}
+		ep := cfg.Config.Entrypoint
```

Comment:
You may also need to consult `cfg.Config.Cmd`, which (AFAIK) should be used if entrypoint isn't set. For example, Tekton does this here: https://github.com/tektoncd/pipeline/blob/302895b5a1ca45c02a85a9822201643c159fe02c/pkg/pod/entrypoint_lookup_impl.go#L163-L166

Reply:
good idea, added